org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/sso/filter/SSOWebUIFilter.java

```diff
@@ -98,10 +99,19 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 					if (m_SSOPrincipal.hasAuthenticationCode(httpRequest, httpResponse))
 					{
 						// Use authentication code get get token
+						String currentUri = httpRequest.getRequestURL().toString();
 						m_SSOPrincipal.getAuthenticationToken(httpRequest, httpResponse, SSOUtils.SSO_MODE_WEBUI);
+
+						// Redirect to default request URL after authentication and handle zoom. 
+						Object zoomPara = httpRequest.getSession().getAttribute(ISSOPrincipalService.SSO_ZOOM_PARAM);
+						if (zoomPara != null && !Util.isEmpty((String) zoomPara))
+							currentUri += "?" + (String) zoomPara;
+						httpResponse.sendRedirect(currentUri);
+						httpRequest.getSession().removeAttribute(ISSOPrincipalService.SSO_ZOOM_PARAM);
 					}
 					else if (!m_SSOPrincipal.isAuthenticated(httpRequest, httpResponse))
 					{
+						httpRequest.getSession().setAttribute(ISSOPrincipalService.SSO_ZOOM_PARAM, httpRequest.getQueryString());
```

Comment:
SSO_ZOOM_PARAM is bad naming here, should use "SSO_QUERY_STRING" instead since it can be command other than zoom.